### PR TITLE
Vested account transfers from a pool sub account to the parent general account

### LINKED ISSUE #11263: [vAMM] allow transfers from a pool derived key to the parent key
Add an optional from field for the transaferFund command.
Requires the key to be:
- a valid subaccount for the main key
- from account to be REWARD_VESTED
- asset to be the same as the transfer

### PR DESCRIPTION
**Pull request by karlem**
Close #11263

### CODE REVIEW

File: protos/sources/vega/commands/v1/commands.proto [hunk 1]
```diff
@@ -304,6 +304,12 @@ message Transfer {
   string amount = 5;
   // Reference to be attached to the transfer.
   string reference = 6;
+  // Optional derived key from which assets are transferred. 
+  // The submitter of the transaction must be the owner of this derived key.
```
Comment (candida-d) [inline protos/sources/vega/commands/v1/commands.proto:308]:
```suggestion
  // Optional AMM key from which assets are transferred. 
  // The submitter of the transaction must be the owner of this AMM key.
```

File: core/banking/transfers_common.go [hunk 2]
```diff
@@ -160,12 +166,17 @@ func (e *Engine) processTransfer(
...
 		return nil, fmt.Errorf("could not pay the fee for transfer: %w", err)
 	}
 	feeTransferAccountType := []types.AccountType{fromAcc}
 
+	// transfer from sub account to owners general account
+	if fromDerivedKey != nil {
+		from = *fromDerivedKey
+	}
+
 	fromTransfer, toTransfer := e.makeTransfers(from, to, assetType.ID, "", toMarket, amount, &transferID)
```
Comment (wwestgarth) [inline core/banking/transfers_common.go:180]: Where do we check that derived key is "owned" by the submitter of the transaction?

Comment (karlem) [reply core/banking/transfers_common.go:180]: I am still working on it. The PR is not done. But almost there!

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 3]
```diff
@@ -4705,6 +4705,9 @@ message EstimateTransferFeeRequest {
   string amount = 4;
   // Asset ID for the asset associated with the transaction.
   string asset_id = 5;
+  // Optional saction must be the owner of this AMM key key from which assets are transferred.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:4708]: What is this line meant to say?

Comment (candida-d) [reply protos/sources/data-node/api/v2/trading_data.proto:4708]: @karlem We might be looking at different lines - I'm looking at " Optional saction must be the owner of this AMM key key from which assets are transferred."

Comment (karlem) [reply protos/sources/data-node/api/v2/trading_data.proto:4708]: Yes. Could you please help me fix it. it sounds like rubbish sorry haha

Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:4708]:
```suggestion
  // AMM public key, if assets are being transferred from an AMM key to the AMM owner's public key.
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 4]
```diff
@@ -4705,6 +4705,9 @@ message EstimateTransferFeeRequest {
...
+  // The `from_account` must be the owner of this AMM key.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:4709]:
```suggestion
  // The `from_account` is the owner of this AMM key.
```

File: protos/sources/data-node/api/v2/trading_data.proto (hunk 3 above)
Comment (candida-d) [reply protos/sources/data-node/api/v2/trading_data.proto:4708]: @karlem Ok - I made a suggestion based on what I think it's doing here

File: protos/sources/vega/commands/v1/commands.proto [hunk 5]
```diff
@@ -304,6 +304,12 @@ message Transfer {
   string amount = 5;
   // Reference to be attached to the transfer.
   string reference = 6;
+  // Optional AMM key from which assets are transferred.
```
Comment (candida-d) [inline protos/sources/vega/commands/v1/commands.proto:307]:
```suggestion
  // AMM key from which assets are to be transferred, if applicable.
```

**Review by wwestgarth — DISMISSED** [review]

**Review by wwestgarth — APPROVED** [review]

**Pull request merged**